src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -65,96 +65,99 @@ const SearchAndFilter: React.FC<filterProps> = ({allData, columns, filterTypes})
-        values.push(value)
+    setFoundFilterIds(resultIds)
+  }, [allData, someFiltersAreSelected, filterTypes, selectedFilterValues])
+
+
+  const makeDataSearchFriendly = (item: object):string[] => {
+    const oneGroupOfValues = Object.values(item).reduce((values:any[], value) => {
+      try {
+        if (Array.isArray(value)) {
+          values.concat(value)
+        } else if (typeof(value) === 'object') {
+          throw `Some of the search results were ignored because the data was too complex`
+        } else {
+          values.push(value)
+        }
+      } catch (e) {
+        console.log(e)
       }
       return values
     }, [])
+
+    return oneGroupOfValues.map(value => value.toUpperCase())
   }
 
   const searchData = () => {
     let result:any[] = [];
 
     if (searchTextBox !== '') {
-      result = allData.map(item => {
-        let itemsValues:any[] = cleanDataForSearch(item)
-        itemsValues = itemsValues.map(value => `${value}`.toUpperCase())
-
-        if (itemsValues.some(value => value.includes(searchTextBox.toUpperCase()))) return item.id
+      result = allData.reduce((results:string[], item) => {
+        let itemsValues:string[] = makeDataSearchFriendly(item)
+        if (itemsValues.some(value => value.includes(searchTextBox.toUpperCase()))) {
+          results.push(item.id)
         }
-      )}
-      result = result.filter(item => item !== undefined)
+        return results
+      }, [])
+    }
 
-    setMatchingSearchIds(result)
+    setFoundSearchIds(result)
   } 
 
-  const determineAvailableFilterValues = (filterType: string) => {
-    const allOptions = allData.reduce((options:(string | number | boolean)[], item:stringKeyOptions): (string | number | boolean)[] => {
-      try {
-        if (Array.isArray(item[filterType])) {
-          options = options.concat(item[filterType])
-        } else if (typeof item[filterType] === 'object') {
-          throw `Some of the filter options for ${filterType} were ignored because the data was too complex`
-        } else {
-          options.push(item[filterType])
-        }
-      } catch (e) {
-        console.error(e)
-      }
-      return options
-    }, [])
-    return allOptions.sort()
-  }
-
-  const someFiltersAreNotDefault = useCallback(() => {
-    return Object.keys(selectedFilters)
-      .some(filter => selectedFilters[filter] !== 'All')
-  }, [selectedFilters])
-
-  const messageIfSearchFailed = () => {
-    const noResults = 'We were unable to find anything, please try again'
-    if (someFiltersAreNotDefault() && limitedListItems.length === 0) {
+  const messageIfSearchFailed = useCallback(() => {
+    const noResults = 'We were unable to find anything based on your search, please try again'
+    if (someFiltersAreSelected() && queriedListItems.length === 0) {
       setUserMessage(noResults) 
-    } else if (searchTextBox !== '' && limitedListItems.length === 0) {
+    } else if (foundSearchIds && queriedListItems.length === 0) {
```

Comment:
This was the great change to `undefined` signaling not to message the user, since `foundSearchIds` is only undefined if there hasn't been a search, or a search has been cleared.